spec/Schema.spec.js

```diff
@@ -1215,4 +1215,47 @@ describe('Class Level Permissions for requiredAuth', () => {
       done();
     });
   });
+
+  it('required auth test create/get/update/delete with roles (#3753)', (done) => {
+    config.database.loadSchema().then((schema) => {
+      // Just to create a valid class
+      return schema.validateObject('Stuff', {foo: 'bar'});
+    }).then((schema) => {
+      return schema.setPermissions('Stuff', {
+        'find': {
+          'requiresAuthentication': true,
+          'role:admin': true
+        },
+        'create': { 'role:admin': true },
+        'update': { 'role:admin': true },
+        'delete': { 'role:admin': true },
+        'get': {
+          'requiresAuthentication': true,
+          'role:admin': true
+        }
+      });
+    }).then(() => {
+      const user = new Parse.User({
+        username: 'user',
+        password: 'password'
+      })
+      return Parse.User.signUp('user', 'password').then(() => {
```

Comment:
Wouldn't it be a more complete test case to check that a query, before user signs up, returns 0 results?